modules/quantcastIdSystem.js

```diff
@@ -7,11 +7,165 @@
+    const usPrivacyString = uspDataHandler.getConsentData();
+
+    var fpa = storage.getCookie(QUANTCAST_FPA),
+     fpan = '0',
+     now = new Date(),
+     domain = quantcastIdSubmodule.findRootDomain(),
+     et = now.getTime(),
+     tzo = now.getTimezoneOffset(),
+     usPrivacyParamString = "",
+     firstPartyParamStrings, 
+     gdprParamStrings;
+
+    if(!hasConsent()) {
+      var expired = new Date(0).toUTCString();
+      fpan = 'u';
+      fpa = '';
+      storage.setCookie(QUANTCAST_FPA, fpa, expired, '/', domain, null);
+    } else if (!fpa) {
+      var expires = new Date(now.getTime() + (cookieExpTime * 86400000)).toGMTString();
+      fpa = 'B0-' + Math.round(Math.random() * 2147483647) + '-' + et;
+      fpan = '1';
+      storage.setCookie(QUANTCAST_FPA, fpa, expires, '/', domain, null);
+    }
+
+    firstPartyParamStrings = `&fpan=${fpan}&fpa=${fpa}`;
+    gdprParamStrings = '&gdpr=0';
+    if (!gdprPrivacyString || typeof gdprPrivacyString.gdprApplies !== 'boolean' || !gdprPrivacyString.gdprApplies) {
+      gdprParamStrings = `gdpr=1&gdpr_consent=${gdprPrivacyString.consentString}`;
+    }
+    if (usPrivacyString && typeof usPrivacyString === 'string') {
+      usPrivacyParamString = `&us_privacy=${usPrivacyString}`;
+    }
+
+    let url = DOMAIN_QSERVE +
+    '?d=' + domain +
+    '&et=' + et +
+    '&tzo=' + tzo +
+    '&client_id=' + clientId +
+    '&a=' + PREBID_PCODE +
+    usPrivacyParamString +
+    gdprParamStrings +
+    firstPartyParamStrings;
+
+    triggerPixel(url);
+  }
+};
+
+function hasConsent() {
+  return hasGDPRConsent() && hasCCPAConsent();
+}    
+
+function hasGDPRConsent() {
+  const gdprConsent = gdprDataHandler.getConsentData();
+
+  // Check for GDPR consent for purpose 1, and drop request if consent has not been given
+  // Remaining consent checks are performed server-side.
+  if (gdprConsent && typeof gdprConsent.gdprApplies === 'boolean' && gdprConsent.gdprApplies) {
+    if (!gdprConsent.vendorData) {
+      return false;
+    }
+    if (gdprConsent.apiVersion === 1) {
+      return checkTCFv1(gdprConsent.vendorData);
+    }
+    if (gdprConsent.apiVersion === 2) {
+      return checkTCFv2(gdprConsent.vendorData);
+    }
+  }
+  return true;
+}
+
+function checkTCFv1(vendorData) {
+  let vendorConsent = vendorData.vendorConsents && vendorData.vendorConsents[QUANTCAST_VENDOR_ID];
+  let purposeConsent = vendorData.purposeConsents && vendorData.purposeConsents[PURPOSE_DATA_COLLECT] ;
+
+  return !!(vendorConsent && purposeConsent);
+}
+
+function checkTCFv2(tcData) {
+  let vendorConsent = tcData.vendor && tcData.vendor.consents && tcData.vendor.consents[QUANTCAST_VENDOR_ID];
+  let vendorInterest = vendorData.vendor.legitimateInterests && vendorData.vendor.legitimateInterests[QUANTCAST_VENDOR_ID];
+  let restrictions = tcData.publisher ? tcData.publisher.restrictions : {};
+
+  //Restrictions for purpose 1
+  let qcRestriction = restrictions && restrictions[PURPOSE_DATA_COLLECT]
```

Comment:
I can probably club the check for Purpose 1 and 10 together like Quantjs does with a `QC_TCF_CONSENT_ONLY_PURPOSE` flag for purpose 1.